src/templates/plugin.main.js

```diff
@@ -90,10 +90,22 @@ export default async (context) => {
     // Lazy-loading enabled
     if (lazy) {
       const { loadLanguageAsync } = require('./utils')
-
-      // Load fallback locale.
-      if (app.i18n.fallbackLocale && newLocale !== app.i18n.fallbackLocale) {
-        await loadLanguageAsync(context, app.i18n.fallbackLocale)
+      const i18nFallbackLocale = app.i18n.fallbackLocale
+
+      // Load fallback locale(s).
+      if (i18nFallbackLocale) {
+        if (typeof i18nFallbackLocale === 'object' && i18nFallbackLocale[newLocale]) {
```

Comment:
`typeof []` is also an `object`.

Since we should also support an array of strings, I think the logic should be:

```js
        if (Array.isArray(i18nFallbackLocale)) {
          await Promise.all(i18nFallbackLocale.map(locale => await loadLanguageAsync(context, locale)))
        } else if (typeof i18nFallbackLocale === 'object' && i18nFallbackLocale[newLocale]) {
```

Also, I'm wondering if we should explicitly avoid loading the currently loaded locale. It seems like we will do an extra `app.i18n.setLocaleMessage` call in that case so it would be better to avoid it.

Reply:
Ok, I've fixed the Array/Object question. Actually, I had forgotten the Array case. I've added a new unit test and the business code to make it green.

For the extra call to `app.i18n.setLocaleMessage`, I think this won't happen (but I may be wrong), as there is an `if` that skips the loading if already loaded:

```js
if (!app.i18n.loadedLanguages) {
    app.i18n.loadedLanguages = []
  }
```

Or did I miss something?

Reply:
> For the extra call to `app.i18n.setLocaleMessage`, I think this won't happen (but I may be wrong), as there is an `if` that skips the loading if already loaded:

Yes but for the initial (SSR) request, when the client-side run this code, the languages will be available through NUXT_STATE but not present in `loadedLanguages`. So then there will be some redundant churn, I believe.

Maybe the original code should be more consistent and set `loadedLanguages` in this case, without actually triggering loading.

Reply:
I must admit I have absolutely no knowledge of the internals of Nuxt.  Do you think this point should be included in this PR that is about fallback, or should it be part of another issue? I'm not even sure about the differences about running the code in SSR and in CSR and its subtilities.

Reply:
You can ignore it in this PR. I'll have a look after this is merged.